packages/core-data/src/fetch/__experimental-fetch-link-suggestions.js

```diff
@@ -153,6 +153,33 @@ const fetchLinkSuggestions = async (
 		);
 	}
 
+	if (
+		type &&
+		type !== 'post' &&
+		type !== 'term' &&
+		type !== 'post-format'
+	) {
+		queries.push(
+			apiFetch( {
+				path: addQueryArgs( '/__experimental/menu-custom-items', {
+					type,
+				} ),
+			} ).then( ( results ) => {
+				const customItems = results.filter(
+					( result ) =>
+						search === '' ||
+						result.title.match( new RegExp( search, 'i' ) )
```

Comment:
This means `search` can contain regex specific tokens such as `*` and `.`, which I don't think is what we expected. If we only want to match case-insensitivity then we can probably just use:

```js
result.title.toLowerCase() === search.toLowerCase()
```